app/models/article.rb

```diff
@@ -827,4 +831,12 @@ def detect_animated_images
 
     ::Articles::DetectAnimatedImagesWorker.perform_async(id)
   end
+
+  def delete_images
+    image_paths = body_markdown.scan(IMG_MARKDOWN_REGEX).flatten
+    main_image_path = main_image&.scan(IMG_MARKDOWN_REGEX)
+    image_paths << main_image_path if main_image_path
+    Images::DeleteWorker.perform_async(image_paths)
```

Comment:
Just thinking out loud: I wonder if it's better to limit each worker to only delete 1 image. Currently, it looks like we are throwing all images of an article to an array. If a worker failed and set to rerun, it would be trying to re-delete all the images again. 

I don't think there's a huge performance difference here but there's a benefit to being precise and strict with 1 arg instead of an array. Thoughts @rhymes ?

Reply:
Hi @maestromac, great point! Having idempotent workers is usually much better. If an article has to delete 5 images and the third fails, we'd be retrying in the first one. Given that the worker is super low priority, I'd rather have 5 workers each deleting one image.

Sorry to ask you this @squarebat but could you alter the code so that the worker only takes one path and instead submits N workers to the queue?

Reply:
Yeah sure! That's actually a great idea.